### PR TITLE
Support for Multiremote and fixed many errors of current main branch

### PR DESCRIPTION
**Pull request by SnakeDrak**
I have added support for multiremote and custom driver when we use the elements or mock matchers. Also I have added some fixes for the current main branch, however the `npm run test:types` is causing issues yet due to external library types. I will fix them in another PR.

It should fix or use as workaround for [issue 106](https://github.com/webdriverio/expect-webdriverio/issues/106).

**UPDATED:** Types fixed as well.

### CODE REVIEW

File: test-types/jest/tsconfig.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
 {
   "compilerOptions": {
     "outDir": "dist",
-    "lib": ["ES2018"],
+    "lib": ["ES2018", "DOM"],
```
Comment (christian-bromann) [inline test-types/jest/tsconfig.json:4]: I don't think we need this.

**Review by christian-bromann — COMMENTED** [review]
Thank you @SnakeDrak for the work on this and sorry for the late reply.

I just pushed a commit that might help avoid all the `@ts-ignore` comments in the PR. I rather would like to avoid them.

**Comment by christian-bromann** [comment]
Also can you explain how this patch will enable multiremote to work?

**Comment by SnakeDrak** [comment]
@christian-bromann I did it some month ago and I don't remember the full details. However, the original idea was to be able to pass the driver where you want to check. You could use something like:

```typescript
expect(myInput).toHaveAttribute('class', 'form-control', browser)
expect(myInput).toHaveAttribute('class', 'md-form-control', app)
```

I remember the master branch was broken and I tried to fix it as well, for that reason you see some extra commits. At this moment, I am using my fork with this feature to support multiples driver at the same instance and to be able to test correctly each one.

You can check the tests included to know if it has a sense for you. Most probably I would have to update against master again.

**Comment by christian-bromann** [comment]
@SnakeDrak is there any interest to push this PR forward?

**Comment by pwespi** [comment]
Is there anything I can do to help pushing this forward? I would like to see the multiremote feature land in expect-webdriverio.

**Comment by lacell75** [comment]
Any news?

**Comment by christian-bromann** [comment]
@lacell75 as you can see, no. Would you like to pick this up?

**Comment by christian-bromann** [comment]
I will go ahead and close this as the branch starts to diverge to much from `main`. I would be happy for any contributions making this library compatible with multiremote.

Thanks for the work you put into this @SnakeDrak

**Pull request closed without merging**